### PR TITLE
Fix checks in PR for empty commits (#20290)

### LINKED ISSUE #19603: Not able to merge commit in PR when branches content is same, but different commit id
### Description

Hi,
there is missing button for commit in pull request, when branches are same but there is commit to merge.
steps to reproduce:

1) "**merge --no-ff**"  from "develop" to "master"
2) create tag on master
3) create PR from "master" to "develop"
4) missing button for merge
PR says the files diff is empty (that is correct).
But also **in PR is 1 commit** with the merge info and created tag, which should be merged into "develop"
(just usual git flow)

### Gitea Version

1.16.7

### Can you reproduce the bug on the Gitea demo site?

Yes

https://try.gitea.io/jedi7/test_19603/pulls/1

### Log Gist

_No response_

### Git Version

2.25.1

### Operating System

ubuntu

### How are you running Gitea?

docker container

### Database

MySQL

### PR DESCRIPTION
**Pull request by jedi7**
backport of PR https://github.com/go-gitea/gitea/pull/20290

* Fix #19603
* fill HeadCommitID in PullRequest
* compare real commits ID as check for merging

### CODE REVIEW

**Review by a1012112796 — APPROVED** [review]

**Comment by 6543** [comment]
@jedi7 tests fail

File: templates/repo/issue/view_content/pull.tmpl [hunk 1]
```diff
@@ -195,12 +195,12 @@
 					<i class="icon icon-octicon">{{svg "octicon-sync"}}</i>
 					{{$.i18n.Tr "repo.pulls.is_checking"}}
 				</div>
-			{{else if .Issue.PullRequest.IsEmpty}}
+			{{else if .Issue.PullRequest.IsAncestor}}
 				<div class="item">
 					<i class="icon icon-octicon">{{svg "octicon-alert" 16}}</i>
-					{{$.i18n.Tr "repo.pulls.is_empty"}}
+					{{$.i18.Tr "repo.pulls.is_ancestor"}}
```
Comment (wxiaoguang) [inline templates/repo/issue/view_content/pull.tmpl]:
```suggestion
					{{$.i18n.Tr "repo.pulls.is_ancestor"}}
```

File: templates/repo/issue/view_content/pull.tmpl [hunk 2]
```diff
@@ -321,6 +320,14 @@
 						</div>
 					</div>
 				{{end}}
+				{{if .Issue.PullRequest.IsEmpty}}
+					<div class="ui divider"></div>
+
+					<div class="item">
+						<i class="icon icon-octicon">{{svg "octicon-alert" 16}}</i>
+						{{$.locale.Tr "repo.pulls.is_empty"}}
```
Comment (wxiaoguang) [inline templates/repo/issue/view_content/pull.tmpl]:
```suggestion
						{{$.i18n.Tr "repo.pulls.is_empty"}}
```

**Review by 6543 — APPROVED** [review]

**Comment by jedi7** [comment]
@wxiaoguang  thank you for fixes :)

**Pull request merged**